src/procedures/__tests__/PushDividendPayment.ts

```diff
@@ -0,0 +1,312 @@
+          ],
+        })
+      );
+
+      wrappersMock.mock(
+        'getAttachedModules',
+        Promise.resolve([erc20DividendsMock.getMockInstance()])
+      );
+
+      const addTransactionSpy = spy(target, 'addTransaction');
+
+      // Real call
+      await target.prepareTransactions();
+
+      // Verifications
+      expect(
+        addTransactionSpy
+          .getCall(0)
+          .calledWith(erc20DividendsMock.getMockInstance().pushDividendPaymentToAddresses)
+      ).toEqual(true);
+      expect(addTransactionSpy.callCount).toEqual(1);
+    });
+
+    test('should add a transaction to push a dividend payment of an attached ERC20 dividends module with undefined shareholderAddresses parameter', async () => {
+      // Instantiate PushDividendPayment
+      target = new PushDividendPayment(
+        {
+          ...params,
+          shareholderAddresses: undefined,
+          dividendType: DividendType.Erc20,
+        },
+        contextMock.getMockInstance()
+      );
+
+      const shareholders = [];
+      for (let i = 0; i < 105; i++) {
+        shareholders.push({
+          address: `0x${Math.floor(Math.random() * 999 + 1)}`,
+          paymentReceived: false,
+        });
+      }
+
+      wrappersMock.mock(
+        'getDividend',
+        Promise.resolve({
+          shareholders,
+        })
+      );
+
+      wrappersMock.mock(
+        'getAttachedModules',
+        Promise.resolve([erc20DividendsMock.getMockInstance()])
+      );
+
+      const addTransactionSpy = spy(target, 'addTransaction');
+
+      // Real call
+      await target.prepareTransactions();
+
+      // Verifications
+      expect(
+        addTransactionSpy
+          .getCall(0)
+          .calledWith(erc20DividendsMock.getMockInstance().pushDividendPaymentToAddresses)
+      ).toEqual(true);
+      expect(addTransactionSpy.callCount).toEqual(2);
+    });
+
+    test('should add a transaction to push a dividend payment of an attached ETH dividends module', async () => {
+      // Instantiate PushDividendPayment
+      target = new PushDividendPayment(
+        { ...params, dividendType: DividendType.Eth },
+        contextMock.getMockInstance()
+      );
+
+      wrappersMock.mock(
+        'getDividend',
+        Promise.resolve({
+          shareholders: [
+            { address: '0x01', paymentReceived: false },
+            { address: '0x02', paymentReceived: false },
+            { address: '0x03', paymentReceived: true },
+          ],
+        })
+      );
+
+      wrappersMock.mock(
+        'getAttachedModules',
+        Promise.resolve([ethDividendsMock.getMockInstance()])
+      );
+
+      const addTransactionSpy = spy(target, 'addTransaction');
+
+      // Real call
+      await target.prepareTransactions();
+
+      // Verifications
+      expect(
+        addTransactionSpy
+          .getCall(0)
+          .calledWith(ethDividendsMock.getMockInstance().pushDividendPaymentToAddresses)
```

Comment:
Same as above (but eth instead of erc20)

This is now only partially correct. You are missing
ethDividendsMock.mock('pushDividendPaymentToAddresses', Promise.resolve('PushDividendPaymentToAddresses')); -> added somewhere before target.prepareTransactions

Where the resolve has a unique string. This is as per what I mentioned in slack today, let me know if you have more questions.

To test this, as the file is right now, you could change ethDividendsMock on this line to erc20DividendsMock, and it would still pass. That is because without the mock. the functions are all undefined type. If you add the code I mention above, you will see that you are no longer able to pass anything else into the calledWith, only this.